app/Http/Controllers/CityOpinionController.php

```diff
@@ -21,13 +21,20 @@ public function store(CityOpinionRequest $request): void
     public function update(CityOpinionRequest $request, CityOpinion $cityOpinion): void
     {
         $opinion = $request->only(["rating", "content", "city_id"]);
-        $opinion["user_id"] = Auth::id();
 
-        $cityOpinion->update($opinion);
+        if ($cityOpinion->user_id === Auth::id()) {
+            $cityOpinion->update($opinion);
```

Comment:
```suggestion
            $cityOpinion->update($request->validated());
```
Instead of `$request->only(["rating", "content", "city_id"]);`

Reply:
Please make policy for that - https://laravel.com/docs/10.x/authorization#creating-policies